src/Game.js

```diff
@@ -0,0 +1,213 @@
+  };
+
+  createPlayer = (username, id, x, y) => {
+    const player = {
+      username: username,
+      id: id,
+      score: 0,
+      paddle: [x, y],
+    };
+    if (this.state.player1) {
+      this.setState({
+        player2: player,
+      });
+    } else {
+      this.setState({
+        player1: player,
+      });
+    }
+  };
+
+  createGameSettings = (x, y) => {
+    const pongBall = [x, y];
+    this.setState({ pongBall: pongBall });
+    // setGameCode(gameCode);
+  };
+
+  createGame = () => {
+    this.createGameSettings(250, 100);
+    this.createPlayer("Jon", 1, 235, 480);
+    this.createPlayer("Joe", 2, 235, 10);
+  };
+
+  startGame = () => {
+    this.createGame();
+    requestAnimationFrame(this.updateAnimationState);
+  };
+
+  animateScene = () => {
+    this.detectCollision();
+    this.incrementBallPosition();
+  };
+
+  incrementBallPosition = () => {
+    this.setState((prevState) => ({
+      pongBall: [
+        prevState.pongBall[0] + prevState.ballDirection[0],
+        prevState.pongBall[1] + prevState.ballDirection[1],
+      ],
+    }));
+  };
+
+  handleKeydown = (event) => {
+    event.preventDefault();
+    const key = event.key;
+    if (key === "ArrowLeft") {
+      this.setState((prevState) => ({
+        player1: {
+          ...prevState.player1,
+          paddle: [
+            prevState.player1.paddle[0] - 15,
+            prevState.player1.paddle[1],
+          ],
+        },
+      }));
+    } else if (key === "ArrowRight") {
+      this.setState((prevState) => ({
+        player1: {
+          ...prevState.player1,
+          paddle: [
+            prevState.player1.paddle[0] + 15,
+            prevState.player1.paddle[1],
+          ],
+        },
+      }));
+    } else if (key === "z") {
+      this.setState((prevState) => ({
+        player2: {
+          ...prevState.player2,
+          paddle: [
+            prevState.player2.paddle[0] - 15,
+            prevState.player2.paddle[1],
+          ],
+        },
+      }));
+    } else if (key === "x") {
+      this.setState((prevState) => ({
+        player2: {
+          ...prevState.player2,
+          paddle: [
+            prevState.player2.paddle[0] + 15,
+            prevState.player2.paddle[1],
+          ],
+        },
+      }));
+    }
+  };
+
+  detectCollision = () => {
+    let [player1PaddleX, player1PaddleY] = this.state.player1.paddle;
+    let [player2PaddleX, player2PaddleY] = this.state.player2.paddle;
+    let [ballX, ballY] = this.state.pongBall;
+
+    let player1RightEdge = player1PaddleX + 30;
+    let player1LeftEdge = player1PaddleX;
+    let player1PaddleTop = player1PaddleY;
+    let player1PaddleBottom = player1PaddleY + 10;
+
+    let player2RightEdge = player2PaddleX + 30;
+    let player2LeftEdge = player2PaddleX;
+    let player2PaddleTop = player2PaddleY + 10;
+    let player2PaddleBottom = player2PaddleY;
+
+    let surface;
+
+    if (
+      (ballX - 10 <= player1RightEdge &&
```

Comment:
This condition should be pulled out in to a variable for readability. Same for other conditions in the if/else chain